src/Featurizers/StandardScaleWrapperFeaturizer.h

```diff
@@ -0,0 +1,348 @@
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+
+// ----------------------------------------------------------------------
+// |
+// |  StandardScalerTransformer
+// |
+// ----------------------------------------------------------------------
+template <typename InputT, typename TransformedT>
+StandardScalerTransformer<InputT, TransformedT>::StandardScalerTransformer(std::double_t average, std::double_t deviation) :
+    _average(std::move(average)),
+    _deviation(std::move(deviation)) {
+        if(_deviation < 0) {
+            throw std::invalid_argument("Standard deviation should be greater or equal to 0 in StandardScaleWrapperFeaturizer!");
+        }
+}
+
+template <typename InputT, typename TransformedT>
+StandardScalerTransformer<InputT, TransformedT>::StandardScalerTransformer(Archive &ar) :
+    StandardScalerTransformer(
+        [&ar](void) {
+            std::double_t average(Traits<std::double_t>::deserialize(ar));
+            std::double_t deviation(Traits<std::double_t>::deserialize(ar));
+
+            return StandardScalerTransformer<InputT, TransformedT>(std::move(average), std::move(deviation));
+        }()
+    ) {
+}
+
+template <typename InputT, typename TransformedT>
+void StandardScalerTransformer<InputT, TransformedT>::save(Archive &ar) const /*override*/ {
+    Traits<std::double_t>::serialize(ar, _average);
+    Traits<std::double_t>::serialize(ar, _deviation);
+}
+
+template <typename InputT, typename TransformedT>
+bool StandardScalerTransformer<InputT, TransformedT>::operator==(StandardScalerTransformer const &other) const {
+
+#if (defined __clang__)
+#   pragma clang diagnostic push
+#   pragma clang diagnostic ignored "-Wfloat-equal"
+#endif
+
+    return this->_average == other._average
+        && this->_deviation == other._deviation;
+
+#if (defined __clang__)
+#   pragma clang diagnostic pop
+#endif
+
+}
+
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+template <typename InputT, typename TransformedT>
+void StandardScalerTransformer<InputT, TransformedT>::execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) /*override*/ {
+    // ----------------------------------------------------------------------
+    using InputTraits                       = Traits<InputT>;
+    using TransformedTraits                 = Traits<TransformedT>;
+    // ----------------------------------------------------------------------
+
+    // if input is Nan for numeric types, it will be treated as missing value
+    // and return a Nan
+    if(InputTraits::IsNull(input)) {
```

Comment:
I don't think that we can make this assumption. There are scenarios where the caller might want to do this and other scenarios where null/NaN values are considered errors.